### PR TITLE
[DOC] clarification how shapes are applied in kmer-hash

### PR DESCRIPTION
**Pull request by SGSSGene**
This adds some clarification on the direction of how shapes are applied when computing hashes.
This intends to solve #2832

### CODE REVIEW

File: include/seqan3/search/views/kmer_hash.hpp [hunk 1]
```diff
@@ -774,6 +774,9 @@ namespace seqan3::views
  *
  * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
  *
+ * \attention The Shape is defined is from right to left. The mask 0b1101 applied to ACGT will return
+ * the sequence AGT.
```
Comment (eseiler) [inline include/seqan3/search/views/kmer_hash.hpp:778]:
```suggestion
 * \attention The Shape is defined from right to left. The mask 0b1101 applied to ACGT will return
 * the sequence AGT.
```

File: test/snippet/search/views/kmer_hash.cpp [hunk 2]
```diff
@@ -14,4 +14,13 @@ int main()
     seqan3::debug_stream << (text | seqan3::views::kmer_hash(seqan3::ungapped{3})) << '\n'; // [6,27,44,50,9]
 
     seqan3::debug_stream << (text | seqan3::views::kmer_hash(0b101_shape)) << '\n'; // [2,7,8,14,1]
+
+    // Note: the Shape is defined is from right to left. The mask 0b1101 applied to ACGT will give
+    // the same result as mask 0b111 applied to AGT.
```
Comment (eseiler) [inline test/snippet/search/views/kmer_hash.cpp:19]:
```suggestion
    // Note: the Shape is defined from right to left. The mask 0b1101 applied to ACGT will yield
    // the same result as mask 0b111 applied to AGT.
```

**Review by eseiler — APPROVED** [review]
👍 
But let's see what we resolve in the dev meeting

File: include/seqan3/search/views/kmer_hash.hpp [hunk 3]
```diff
@@ -774,6 +774,9 @@ namespace seqan3::views
  *
  * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
  *
+ * \attention The Shape is defined from right to left. The mask 0b1101 applied to ACGT will return
+ * the sequence AGT.
```
Comment (smehringer) [inline include/seqan3/search/views/kmer_hash.hpp:778]: why is this an `attention`? Isn't this intuitive to read the shape from left to right?

Also:
```suggestion
 * \note The Shape is defined from right to left. The mask 0b1101 applied to "ACGT" is 
 * interpreted as "A.GT" will return the hash value for "AGT".
```

**Review by smehringer — CHANGES_REQUESTED** [review]
small :nail_care:

File: test/snippet/search/views/kmer_hash.cpp [hunk 4]
```diff
@@ -14,4 +14,13 @@ int main()
...
     seqan3::debug_stream << (text | seqan3::views::kmer_hash(0b101_shape)) << '\n'; // [2,7,8,14,1]
+
+    // Note: the Shape is defined from right to left. The mask 0b1101 applied to ACGT will yield
+    // the same result as mask 0b111 applied to AGT.
+    {
+        auto text1 = "ACGT"_dna4;
+        auto text2 = "AGT"_dna4;
+        seqan3::debug_stream << (text1 | seqan3::views::kmer_hash(0b1101_shape)) << '\n'; // [11]
+        seqan3::debug_stream << (text2 | seqan3::views::kmer_hash(0b111_shape)) << '\n';  // [11]
+    }
```
Comment (smehringer) [inline test/snippet/search/views/kmer_hash.cpp:25]:
```suggestion
    // Attention: the Shape is defined from right to left! 
    // The mask 0b11111101 applied to "AGAAAATA" ("A.AAAATA") will yield
    // the same hash value as mask 0b1111111 applied to "AAAAATA".
    {
        auto text1 = "AGAAAATA"_dna4;
        auto text2 = "AAAAATA"_dna4;
        seqan3::debug_stream << (text1 | seqan3::views::kmer_hash(0b11111101_shape)) << '\n'; // [TODO?]
        seqan3::debug_stream << (text2 | seqan3::views::kmer_hash(0b1111111_shape)) << '\n';  // [TODO?]
    }
```

**Review by smehringer — CHANGES_REQUESTED** [review]

**Review by smehringer — APPROVED** [review]

**Pull request merged**